src/profile/helpers/editData.ts

```diff
@@ -150,17 +182,254 @@ function pickSources(
+  if (index < 0) {
+    throw new Error('Item not found in updateItemAndCloneList() ');
+  }
+  const updatedItem = cloneAndMutateItem(item, {
+    value: newValue,
+    saving: saving || undefined,
+  });
+  const newList = _.cloneDeep(allItems);
+  newList[index] = updatedItem;
+  return newList;
+}
+
+export function createNewProfileNode<T extends MultiItemProfileNode>(
+  dataType: EditDataType,
+  overrides?: Partial<T>
+): T {
+  return {
+    ...(formConstants.EMPTY_VALUES[dataType] as T),
+    ...overrides,
+  };
+}
+
+export function createFormValues(
+  allItems: EditData[],
+  dataType: EditDataType
+): Partial<FormValues> {
+  if (!isMultiItemDataType(dataType)) {
+    const value = allItems[0].value as
+      | BasicDataValue
+      | AdditionalInformationValue;
+    return {
+      ...value,
+    };
+  } else {
+    const nodes = allItems
+      .filter(item => item.saving !== 'remove')
+      .map(item =>
+        createNewProfileNode(dataType, {
+          ...item.value,
+          primary: item.primary,
+        })
+      );
+    return {
+      [dataType]: nodes,
+    };
+  }
+}
+
+function hasItemUpdated(
+  item: EditData,
+  dataType: EditDataType,
+  profileData: EditDataProfileSource
+): boolean {
+  const { saving } = item;
+  if (!saving) {
+    return false;
+  }
+  if (saving === 'value') {
+    const profileDataValue = pickValue(profileData, dataType) as BasicDataValue;
+    return _.isEqual(item.value, profileDataValue);
+  }
+  if (saving === saveTypeSetPrimary) {
+    return _.isEqual(
+      item.primary,
+      (profileData as MultiItemProfileNode).primary
+    );
+  }
+  return false;
+}
+
+export function updateItems(
+  allItems: EditData[],
+  profileDataItems: EditDataProfileSource[],
+  dataType: EditDataType
+): EditData[] | null {
+  let dataHasUpdated = false;
+  const newList: EditData[] = [];
+  let newItem = getNewItem(allItems);
+  profileDataItems.forEach(profileDataItem => {
+    const existingItem = findItem(allItems, profileDataItem.id);
+    const newValue = pickValue(profileDataItem, dataType);
+    if (existingItem) {
+      const itemHasUpdated = hasItemUpdated(
+        existingItem,
+        dataType,
+        profileDataItem
+      );
+      if (itemHasUpdated) {
+        const copy = createNewItem(profileDataItem, dataType, {
+          saving: undefined,
+        });
+        newList.push(copy);
+        dataHasUpdated = true;
+      } else {
+        newList.push(existingItem);
+      }
+    } else {
+      if (newItem && _.isEqual(newItem.value, newValue)) {
+        dataHasUpdated = true;
+        newItem = undefined;
+      }
+      const copy = createNewItem(profileDataItem, dataType);
+      newList.push(copy);
+    }
+  });
+  if (newItem) {
+    newList.push(newItem);
+  }
+  if (!dataHasUpdated && allItems.length !== profileDataItems.length) {
+    dataHasUpdated = true;
+  }
+  return dataHasUpdated ? newList : null;
+}
+
+export function createBackups(): Backups {
```

Comment:
Did not find any usages outside of this file, so maybe the export is unnecessary?

Reply:
Removed.